python/revng/scripting/projects/cli_project.py

```diff
@@ -0,0 +1,106 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import subprocess
+from tempfile import NamedTemporaryFile
+from typing import List
+
+import yaml
+
+from revng.model import Binary, DiffSet, Function  # type: ignore[attr-defined]
+
+from .project import Project
+
+
+class CLIProject(Project):
+    """
+    This class is used to run revng analysis and artifact through
+    the CLI with subprocess.
+    """
+
+    def __init__(self, resume_dir: str | None = None, revng_executable: str | None = None):
+        self._set_resume_dir(resume_dir)
+        self._set_revng_executable(revng_executable)
+        super().__init__()
```

Comment:
Usually one runs `super().__init__()` first of all.

Reply:
Hmmm, the thing is that in the parent constructor the `_load` function is called, and that assumes that the `resume_dir` and the `revng_executable` are set to a correct value.
Looking again at this I think the best course of action is to make those two constructor parameters

Reply:
Refactored a bit the code, now it should be much easier to follow along (and with top-level `super()`)